WordPress/src/androidTest/java/org/wordpress/android/modules/DummyInterceptorModule.kt

```diff
@@ -0,0 +1,19 @@
+package org.wordpress.android.modules
+
+import dagger.Module
+import dagger.Provides
+import dagger.hilt.components.SingletonComponent
+import dagger.hilt.testing.TestInstallIn
+import dagger.multibindings.IntoSet
+import okhttp3.Interceptor
+import javax.inject.Named
+
+@TestInstallIn(
+        components = [SingletonComponent::class],
+        replaces = [InterceptorModule::class]
+)
+@Module
+class DummyInterceptorModule {
+    @Provides @IntoSet @Named("network-interceptors")
+    fun provideNetworkInterceptor(): Interceptor = Interceptor { it.proceed(it.request()) }
+}
```

Comment:
@hichamboushaba I copied `DummyInterceptorModule` from WCAndroid. It works but I couldn't understand what this actually does. Is `@Named("network-interceptors")` required for fluxc library? Should I check fluxc documents?

Reply:
@irfano I need to check, but if I remember correctly the reason we needed this it was just to replace the actual `InterceptorModule` in tests with a dummy one, so the main thing is:
```kotlin
@TestInstallIn(
        components = [SingletonComponent::class],
        replaces = [InterceptorModule::class]
)
```
and not the interceptor itself, I didn't test this, but maybe you can just keep the module empty, without the provided interceptor at all 🤔

Reply:
Thanks for the explanation.

I learnt that "network-interceptors" is mainly for logging, "interceptors" is for mocking [here](https://github.com/wordpress-mobile/WordPress-FluxC-Android/pull/1155).
WCAndroid is using [Flipper](https://fbflipper.com/), WPAndroid is using [Stetho](http://facebook.github.io/stetho/) for network debugging. By adding DummyInterceptorModule, WCAndroid removed Flipper for testing. I couldn't understand the exact reason why do we need to cancel network debugging, but I will also do the same for WPAndroid.

I mean I removed Stetho for testing by adding DummyInterceptorModule.

Reply:
We remove the logging because it adds an overhead which is not necessary for tests, since the network requests are already mocked, but I don't think it will hurt if it stayed, if it's just pure logging.

Reply:
It makes sense now. Thanks! 👍🏻